### ISSUE TITLE
Add ArrayFire benchmarks

### ISSUE THREAD

**Issue by svenstaro** [boostorg/compute#501]
I think it would be interesting to see [ArrayFire](http://arrayfire.com/) benchmarks especially because it claims to be (mostly) faster than boost.compute in their [benchmarks](http://arrayfire.com/benchmarking-parallel-vector-libraries/).

**Comment by pavanky** [comment]
The benchmark code used for the blog can be seen here: https://github.com/arrayfire/valhalla. I am trying to make it a full blown benchmarking suite for all the libraries used in that blog post. 

@kylelutz If you are thinking about doing this let me know if you need any help from my end.

**Comment by VincentSC** [comment]
Any update on this? And any tips for Boost.Compute how to make it faster?

**Comment by pavanky** [comment]
@VincentSC One of the things I have noticed where ArrayFire is faster is because of the memory manager. If you can modify your local boost compute to implement even a rudimentary memory manager it can improve the performance quite a bit at smaller sizes.

**Comment by kylelutz** [comment]
Yeah, I'm hesitant to add more benchmark applications directly to Boost.Compute. I more like the `valhalla` approach to providing a single, separate benchmark suite which compares a variety of GPU libraries. This seems preferable to each individual library maintaining their own benchmark suite for every other library.

And a memory manager would be good to have, especially for algorithms that could re-use the same "scratch" memory buffers over and over.

**Comment by svenstaro** [comment]
In that case maybe you should link to an external suite of benchmarks instead of providing your own and then work with the maintainer of that benchmark suite?
